src/lib.rs

```diff
@@ -571,7 +571,7 @@ impl<F: Read + Write + Seek> CompoundFile<F> {
     ) -> io::Result<CompoundFile<F>> {
         let mut header = Header {
             version,
-            num_dir_sectors: 1,
+            num_dir_sectors: if version == Version::V3 { 0 } else { 1 },
```

Comment:
Could you add a comment indicating that section 2.2 of the spec requires this field to be zero in version 3?